### PR TITLE
Issue #27

### PR DESCRIPTION
**Pull request by csanders-git**

### CODE REVIEW

File: http.py [hunk 1]
```diff
@@ -46,6 +53,69 @@ def parse_content_encoding(self, response_headers, response_data):
                 })        
         return response_data
 
+    def checkForCookie(self, cookie):
```
Comment (zmallen) [inline http.py]: no camel case here, make it check_for_cookie

File: http.py [hunk 2]
```diff
@@ -46,6 +53,69 @@ def parse_content_encoding(self, response_headers, response_data):
...
+        # http://bayou.io/draft/cookie.domain.html
+        # Check if our originDomain is an IP
+        originIsIP = True
+        try:
+            IP(self.dest_addr)
+        except:
+            originIsIP = False
+        
+        for cookieName, cookieMorsals in cookie.iteritems():
```
Comment (zmallen) [inline http.py]: fix camelcase

File: http.py [hunk 3]
```diff
@@ -46,6 +53,69 @@ def parse_content_encoding(self, response_headers, response_data):
...
+                # http://tools.ietf.org/html/rfc6265#section-4.1.2.3
+                firstNonDot = 0
+                for i in range(len(coverDomain)):
+                    if coverDomain[i] != '.':
+                        firstNonDot = i
+                        break
+                coverDomain = coverDomain[firstNonDot:]
+                # We must parse the coverDomain to make sure its not in the suffix list
+                try:
+                    with open('util/public_suffix_list.dat', 'r') as f:
```
Comment (zmallen) [inline http.py]: move this to a config file/static variable

File: http.py [hunk 4]
```diff
@@ -46,6 +53,69 @@ def parse_content_encoding(self, response_headers, response_data):
...
+        try:
+            IP(self.dest_addr)
+        except:
+            originIsIP = False
+        
+        for cookieName, cookieMorsals in cookie.iteritems():
+            # If the coverdomain is blank or the domain is an IP set the domain to be the origin
+            if cookieMorsals['domain'] == "" or originIsIP is True:
+                # We want to always add a domain so it's easy to parse later
+                return (cookie, self.dest_addr)
```
Comment (zmallen) [inline http.py]: remove parantheses

File: http.py [hunk 5]
```diff
@@ -10,12 +10,19 @@
...
 
 class HttpResponse(object):
-    def __init__(self, http_response):
+    def __init__(self, http_response, user_agent):
         self.response = http_response
+        # For testing purposes HTTPResponse might be called OOL        
+        try:
+            self.dest_addr = user_agent.request_object.dest_addr
+        except AttributeError:
+            self.dest_addr = "127.0.0.1"
```
Comment (zmallen) [inline http.py]: single quotes

File: http.py [hunk 6]
```diff
@@ -46,6 +53,69 @@ def parse_content_encoding(self, response_headers, response_data):
...
+                # if our cover domain is the origin TLD we ignore
+                # Quick sanity check
+                if coverDomain == oTLD:
+                    return False
+                # check if our coverdomain is a subset of our origin domain
+                # Domain match (case insensative)
+                if coverDomain == self.dest_addr:
+                    return (cookie, self.dest_addr)
+                # Domain match algorithm
+                B = coverDomain.lower()
```
Comment (zmallen) [inline http.py]: what is B? descriptive name

File: http.py [hunk 7]
```diff
@@ -46,6 +53,69 @@ def parse_content_encoding(self, response_headers, response_data):
...
+                HDN = self.dest_addr.lower()
```
Comment (zmallen) [inline http.py]: what is HDN? descriptive name

File: http.py [hunk 8]
```diff
@@ -46,6 +53,69 @@ def parse_content_encoding(self, response_headers, response_data):
...
+                NEnd = HDN.find(B)
+                if NEnd is not False:
+                    N = HDN[0:NEnd]
```
Comment (zmallen) [inline http.py]: what is N? descriptive name

File: http.py [hunk 9]
```diff
@@ -71,7 +141,31 @@ def process_response(self):
...
+            except Cookie.CookieError as err:
+                raise errors.TestError(
+                    'Error processing the cookie content into a SimpleCookie',
+                    {
+                        'msg': str(err),
+                        'set_cookie': str(response_headers["set-cookie"]),
+                        'function': 'http.HttpResponse.process_response'
+                    })
+            # if the checkForCookie is invalid then we don't save it
+            if self.checkForCookie(cookie) is False:
```
Comment (zmallen) [inline http.py]: no camelcase

File: http.py [hunk 10]
```diff
@@ -163,7 +259,24 @@ def build_socket(self):
                     'message': msg,
                     'function': 'http.HttpUA.build_socket'
                 })
-
+    def find_cookie(self):
```
Comment (zmallen) [inline http.py]: fix camelcase in this function

File: http.py [hunk 11]
```diff
@@ -163,7 +259,24 @@ def build_socket(self):
...
+        originDomain = self.request_object.dest_addr
+        for cookie in self.cookiejar:
+            for cookieName, cookieMorsals in cookie[0].iteritems():
+                coverDomain = cookieMorsals['domain']
+                if coverDomain == "":
+                    if originDomain == cookie[1]:
+                        returnCookies.append(cookie[0])
+                else:
+                    # Domain match algorithm
+                    B = coverDomain.lower()
```
Comment (zmallen) [inline http.py]: what is B/HDN? descriptive name

File: http.py [hunk 12]
```diff
@@ -174,7 +287,45 @@ def build_request(self):
...
             request, '#version#', self.request_object.version)
-    
+        available_cookies = self.find_cookie()
+        # If the user has requested a tracked cookie and we have one set it
+        if available_cookies:
+            cookie_value = ""
+            if 'cookie' in self.request_object.headers.keys():
+                # Create a SimpleCookie out of our provided cookie
+                try:
+                    cookieX = Cookie.SimpleCookie()
```
Comment (zmallen) [inline http.py]: what is cookieX? descriptive name

File: http.py [hunk 13]
```diff
@@ -163,7 +259,24 @@ def build_socket(self):
...
+                else:
+                    # Domain match algorithm
+                    B = coverDomain.lower()
+                    HDN = originDomain.lower()
+                    NEnd = HDN.find(B)
+                    if NEnd is not False:
+                        returnCookies.append(cookie[0])
+        return returnCookies
+        
     def build_request(self):
```
Comment (zmallen) [inline http.py]: fix camelcase here

**Comment by zmallen** [comment]
Some things here:

1. Can you run PEP8 and fix all the violations? 
2. Make cookie unit tests/integration tests that arent in test_http, combining tests is a bad idea if we dont have tests that go and check the atomic components 
3. Make Travis pass failing test, also add in tests to run in `.travis.yml`

**Comment by zmallen** [comment]
Also try rebasing to see if tests pass, i dont think you merged latest code

**Comment by RyPeck** [comment]
@csanders-git possible to keep the `public_suffix_list.dat` out of the repo and either use the Python module (https://pypi.python.org/pypi/publicsuffixlist) or always pull the latest list from the website?

File: http.py (hunk 3 above)
Comment (csanders-git) [inline http.py]: What are you referring too here?

File: http.py (hunk 4 above)
Comment (csanders-git) [inline http.py]: Which perens...?

File: http.py (hunk 6 above)
Comment (csanders-git) [inline http.py]: this is from the algorithm specified in rfc2965

File: http.py (hunk 7 above)
Comment (csanders-git) [inline http.py]: this is from the algorithm specified in rfc2965

File: http.py (hunk 8 above)
Comment (csanders-git) [inline http.py]: this is from the algorithm specified in rfc2965

**Comment by zmallen** [comment]
This should close out #27

**Comment by zmallen** [comment]
@RyPeck / @csanders-git can you make an issue to move public_suffixes to a git submodule/dynamically pull/use the package?

**Pull request merged**